functions/go/gatekeeper/main.go

```diff
@@ -76,12 +92,96 @@ func (gkp *GatekeeperProcessor) Process(resourceList *framework.ResourceList) er
 	return nil
 }
 
+func (gkp *GatekeeperProcessor) ProcessInput() error {
+	content, err := ioutil.ReadFile(gkp.input)
+	if err != nil {
+		return fmt.Errorf("unable to process input: %w", err)
+	}
+	if gkp.useJSON {
+		content, err = k8syaml.JSONToYAML(content)
+		if err != nil {
+			return fmt.Errorf("unable to process input: %w", err)
+		}
+	}
+	gkp.inputBuf = bytes.NewBuffer(content)
+	return nil
+}
+
+func (gkp *GatekeeperProcessor) ProcessOutput() error {
+	var err error
+	if gkp.outputBuf == nil {
+		return fmt.Errorf("the output buffer must not be nil")
+	}
+	content := gkp.outputBuf.Bytes()
+	if gkp.useJSON {
+		content, err = k8syaml.YAMLToJSON(content)
+		if err != nil {
+			return fmt.Errorf("unable to process output: %w", err)
+		}
+	}
+
+	err = ioutil.WriteFile(gkp.output, content, 0644)
+	if err != nil {
+		return fmt.Errorf("unable to process output: %w", err)
+	}
+	return nil
+}
+
+func (gkp *GatekeeperProcessor) Read(p []byte) (n int, err error) {
+	if gkp.inputBuf == nil {
+		return 0, nil
+	}
+	return gkp.inputBuf.Read(p)
+}
+
+func (gkp *GatekeeperProcessor) Write(p []byte) (n int, err error) {
+	if gkp.outputBuf == nil {
+		gkp.outputBuf = &bytes.Buffer{}
+	}
+	return gkp.outputBuf.Write(p)
+}
+
+func (gkp *GatekeeperProcessor) addFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&gkp.input, "input", "i", stdin,
+		`path to the input file`)
+	cmd.Flags().StringVarP(&gkp.output, "output", "o", stdout,
+		`path to the output file`)
+	cmd.Flags().BoolVar(&gkp.useJSON, "json", false,
+		`input and output is JSON instead of YAML`)
```

Comment:
How do these flags work with the resourceList input provided by the function orchestrator.

Can you pl. describe the change in the PR description.

Reply:
Updated. PTAL.

Reply:
Since we are adding these flags, Can we add a test (script based) to cover these flag's usage ?

Reply:
Added a few tests to exercise the flags